resources/fetchers/kube_provider.go

```diff
@@ -62,19 +65,43 @@ func GetKubeData(watchers []kubernetes.Watcher) []fetching.Resource {
 	return ret
 }
 
-func (r K8sResource) GetID() (string, error) {
+func (r K8sResource) GetData() interface{} {
+	return r.Data
+}
+
+func (r K8sResource) GetMetadata() fetching.ResourceMetadata {
+	k8sObjMeta := r.GetK8sObjectMeta()
+	resourceID := k8sObjMeta.UID
+	resourceName := k8sObjMeta.Name
+
+	return fetching.ResourceMetadata{
+		ID:      string(resourceID),
+		Type:    k8sObjType,
+		SubType: r.GetSubType(),
+		Name:    resourceName,
+	}
+}
+
+func (r K8sResource) GetK8sObjectMeta() metav1.ObjectMeta {
 	k8sObj := reflect.Indirect(reflect.ValueOf(r.Data))
 	metadata, ok := k8sObj.FieldByName(k8sObjMetadataField).Interface().(metav1.ObjectMeta)
 	if !ok {
-		return "", fmt.Errorf("failed to retrieve object metadata")
+		logp.L().Errorf("failed to retrieve object metadata, Resource: %#v", r)
+		return metav1.ObjectMeta{}
 	}
 
-	uid := metadata.UID
-	return string(uid), nil
+	return metadata
 }
 
-func (r K8sResource) GetData() interface{} {
-	return r.Data
+func (r K8sResource) GetSubType() string {
```

Comment:
Is there a reason not to fail and continue with an empty string value instead of the kind?
Just wondering if we are interested in sending data without kind that might create problems down the line of the pipeline that expects to groupBy or do any logic on these values?

Reply:
good question👆🏼
I think i'd prefer events with subtype missing than no events at all,
but the possibility for problems in Kibana resulting from this field missing is interesting...

Reply:
I was thinking about it...
I guess in the UI we can filter those events or just present them last than not sending them.
@eyalkraft, wdyt?

Reply:
I think sending with empty subtype matches the current design pattern and is a good start (best effort).
for example currently [if a fetcher registration fails](https://github.com/elastic/cloudbeat/blob/25c12c588ec6e4841255aa3afa7aeeac7655ac56/resources/manager/factory.go#L69-L71) cloudbeat will log an error and continue.
anyways it's a good idea to ask @kfirpeled / @ari-aviran or someone else from the FE, or even comment [here](https://github.com/elastic/security-team/issues/3681).

Reply:
Additionally, except for bugs I don't see any reason for this method ever failing so I wouldn't waste too much time thinking/debating it.

Reply:
Thats a good point, 
In case of an issue - we can define it to be empty. And we would know to treat empty values as error.

What happens when there's no subtype? For example, in process. 
Just to make things easier, I think it is better to have value we would like to see. And not solve this in the UI only or assume if it is process there's no error and it is empty. Just so we can allow our users to built their own dashboard over the data without having some sort of logic.